ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron/Forge.hs

```diff
@@ -119,15 +145,35 @@ forgeBlock (WithEBBNodeConfig cfg) curSlot curNo prevHash txs isLeader = do
       , pbftProtocolMagic
       } = encNodeConfigExt cfg
 
+    blockPayloads :: BlockPayloads
+    blockPayloads = foldl' extendBlockPayloads initBlockPayloads txs
+
     txPayload :: CC.UTxO.TxPayload
-    txPayload = CC.UTxO.mkTxPayload (map (void . byronTx) txs)
+    txPayload = CC.UTxO.mkTxPayload (bpTxs blockPayloads)
+
+    dlgPayload :: CC.Delegation.Payload
+    dlgPayload = CC.Delegation.unsafePayload (bpDlgCerts blockPayloads)
+
+    updatePayload :: CC.Update.Payload
+    updatePayload = CC.Update.payload (bpUpProposal blockPayloads)
+                                      (bpUpVotes blockPayloads)
+
+    extendBlockPayloads :: BlockPayloads
+                        -> GenTx (ByronBlockOrEBB cfg)
+                        -> BlockPayloads
+    extendBlockPayloads bp@BlockPayloads{bpTxs, bpDlgCerts, bpUpVotes} genTx =
+      case genTx of
+        ByronTx             _ tx   -> bp { bpTxs        = void tx : bpTxs }
+        ByronDlg            _ cert -> bp { bpDlgCerts   = void cert : bpDlgCerts }
+        ByronUpdateProposal _ prop -> bp { bpUpProposal = Just (void prop) }
```

Comment:
Can you add a TODO on this line that we should throw an error in case of multiple update proposals?